### ISSUE TITLE
A connection attempt failed because the connected party did not properly respond after a period of time

### ISSUE THREAD

**Issue by superzadeh** [docker/for-win#2575]
- [X] I have tried with the latest version of my channel (Stable or Edge)
  - [X] I have uploaded Diagnostics
  - Diagnostics ID: BD2EB361-BE29-4A76-9BF5-8D17D0BAE512/20180910105402
  - Diagnostics ID: BD2EB361-BE29-4A76-9BF5-8D17D0BAE512/20180912112953
  - Diagnostics ID: BD2EB361-BE29-4A76-9BF5-8D17D0BAE512/20180920162336
  - Diagnostics ID: BD2EB361-BE29-4A76-9BF5-8D17D0BAE512/20180927141455

### Expected behavior

Docker and Docker Compose commands should execute without error.

### Actual behavior

`docker` commands fail with a message `Error response from daemon: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.`

`docker-compose` commands fail with a message `ERROR: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.`

### Information

  - Windows Version: 10.0.17134
  - Docker for Windows Version: 18.06.1-ce-win73 (19507)
  - Compose Version: 1.22

![image](https://user-images.githubusercontent.com/11708669/45289484-9089d100-b4ed-11e8-84c8-e2f1bd5b8a13.png)

### Steps to reproduce the behavior

Sorry if the steps to reproduce below seems a bit funky, however this is how I reproduce it 100% of the time these days.

  1. Boot your machine, docker starts
  2. Put your machine into your bag (goes into sleep mode), go to work (I have about 30min commute)
  OR Close up your laptop, go out for lunch (I usually take 30min to 1 hour to eat)
  3. Pull out your machine, run a command to start containers ==> run into the issue

### Additional info

The Docker icon in systray seems to indicate that everything is running fine, however it's not. Restarting Docker usually doesn't help, most of the time I need to reboot my laptop.

**Comment by petersedivec** [comment]
Seeing the same issue running (Windows 10 Version 1709 OS Build 16299.15)
![image](https://user-images.githubusercontent.com/6617012/45563918-a526db80-b803-11e8-8827-2d7f73cdf97d.png)

**Comment by superzadeh** [comment]
Any update on this? Is there anything else I can do to help? Run into it multiple times again, uploaded some of the diagnostics.

**Comment by ManfredLange** [comment]
Same problem here. Uploaded two crash reports, then I picked "Reset to defaults" (or whatever it said on the crash report upload dialog box. Although this removes all containers, Docker starts working again without reboot. And I have a script anyways to create all the containers I need. Next time this happens I'll reset to defaults right away.

**Comment by crowne** [comment]
Same issue running on Windows 10 Pro 1803 OS Build 17134.471
This happens when I run docker-compose up -d
Kitematic also fails to connect.

![image](https://user-images.githubusercontent.com/3218673/50134459-b9fa8b80-0298-11e9-852d-8e45e62cad87.png)

**Comment by crowne** [comment]
I think I've resolved my issue by increasing the Memory from 2048 MB to 4096 MB and the Swap 1024 MB to 2048 MB.

![image](https://user-images.githubusercontent.com/3218673/50151348-52a8ff80-02c9-11e9-83d7-844013ea4fa2.png)

**Comment by Karlheinzniebuhr** [comment]
Same issue after updating to version Version **2.0.0.0-win81 (29211) 4271b9e**

**Comment by SrinivasTangella** [comment]
> I think I've resolved my issue by increasing the Memory from 2048 MB to 4096 MB and the Swap 1024 MB to 2048 MB.
> 
> ![image](https://user-images.githubusercontent.com/3218673/50151348-52a8ff80-02c9-11e9-83d7-844013ea4fa2.png)

Yes, this worked!

**Comment by tshepza** [comment]
Iyhoo after a week of struggle  I came across this post. Fixed my issue by increassing Memory to 4096

**Comment by propagandaCreative** [comment]
This worked for me, too

**Comment by eugene-matvejev** [comment]
i tried all options -
```
> docker ps
error during connect: Get https://192.168.99.100:2376/v1.39/containers/json: dial tcp 192.168.99.100:2376: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.```

same error

mac work fine, but windows is don't run at all

**Comment by benbostic** [comment]
Same here, @eugene-matvejev. Tried the resource adjustments. Same error.

Specifically, mine is failing in an Azure DevOps Docker Task (push) on a POST to the Docker host.

**Comment by lolmaus** [comment]
Hi! Using Windows 10 Pro x64 version 1903 (10.0.18362 Build 18362) on AMD Ryzen with Hyper-V enabled.

Kitematic does work in native mode and does run containers:

![image](https://user-images.githubusercontent.com/200884/58160148-8e842e80-7c86-11e9-8e71-94256d23434f.png)

But Docker CLI does not work.

Trying `docker info` in a PowerShell initiated from Kitematic results in a long pause and then an error message: 

> error during connect: Get https://192.168.99.100:2376/v1.39/info: dial tcp 192.168.99.100:2376: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.

Increasing the memory setting in Docker Desktop did not help.

Please advise.

**Comment by a90machado** [comment]
On Windows 10 increasing the Memory and Swap resolve my issue.

**Comment by lolmaus** [comment]
I've noticed that the error message contains a different IP than Docker Desktop is using:

```
$ docker info
error during connect: Get https://192.168.99.100:2376/v1.39/info: dial tcp 192.168.99.100:2376: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
```

<img src="https://user-images.githubusercontent.com/200884/59676187-a8eef080-91cf-11e9-9f50-dd7d93ed554a.png" alt="sceenshot" width="833">

```
$ ipconfig

(...)

Ethernet adapter vEthernet (DockerNAT):

   Connection-specific DNS Suffix  . :
   IPv4 Address. . . . . . . . . . . : 10.0.75.1
   Subnet Mask . . . . . . . . . . . : 255.255.255.0
   Default Gateway . . . . . . . . . :

(...)
```

`ipconfig` is not reporting any adapter with a `192.168.100.x` address.

***

Why is the `docker` CLI command using `192.168.99.100`? How do I make it use `10.0.75.x`?

**Comment by allout58** [comment]
@lolmaus Was running into the same issue, seems to have been an issue with having both Docker Toolbox (old version using VirtualBox) and Docker for Windows (using Hyper-V) installed. Removing Docker Toolbox and re-installing Docker for Windows seems to have fixed it for me.

**Comment by lolmaus** [comment]
Thank you so much, @allout58! This resolved my issue. 🙇 

Note that I also had to [delete environment variables](https://github.com/docker/for-win/issues/1746) starting with `DOCKER` which also explains where the wrong IP address was coming from!

**Comment by jheidlage1222** [comment]
@lolmaus I did both delete those env vars and remove docker toolbox before reinstalling and the issue is resolved.  Much appreciated!

**Comment by JohnLins** [comment]
I had this same problem, you just have to make sure that the VM is running correctly if your using Windows or Mac, Docker depends on Virtual Box running Linux in this case, so you must open Kitematic and click the VirtualBox button to allow access. When I did this, the hello world ran correctly!

**Comment by brakop** [comment]
My issue on this was similar, but not my resolution. I tried removing Docker desktop and using the windows set command from command prompt to remove DOCKER* env variables, but after installing Docker Desktop again the environment variables just came back. I then went into the System properties through the Win10 properties and removed them from there (and did not reinstall Dicker Desktop) and then finally things were fine. This issued started for me because of a Win10 Pro upgrade from Win7 Pro.
